osversion/platform_compat_windows_test.go

```diff
@@ -0,0 +1,61 @@
+package osversion
+
+import (
+	"testing"
+)
+
+const (
+	RS5Prefix         = "10.0.17763"
+	V21H2ServerPrefix = "10.0.20348"
+	V22H2Win11Prefix  = "10.0.22621"
+)
+
+// Test the platform compatibility of the different
+// OS Versions considering two ltsc container image
+// versions (ltsc2019, ltsc2022)
+func Test_PlatformCompat_HostWS2022(t *testing.T) {
+	for _, tc := range map[string]struct {
+		host string
+	}{
+		"PullCtrImage_WS2022Host": {
+			host: V21H2ServerPrefix,
+		},
+		"PullCtrImage_RS5Host": {
+			host: RS5Prefix,
+		},
+		"PullCtrImage_Wind11Host": {
+			host: V22H2Win11Prefix,
+		},
+	} {
+		// Check if ltsc2019/ltsc2022 container images are compatible on
+		// the given host OS versions
+		//
+		switch tc.host {
+		case V21H2ServerPrefix:
+			// only ltsc 2022 ctr image version is compatible on WS 2022
+			if CanRun(tc.host, V21H2ServerPrefix) != true {
```

Comment:
This test might be more go idiomatic if `CanRun` was only called once and the "Prefix" was part of the testcase above.  something like:

```
if CanRun(tc.host, tc.guestOs) != tc.shouldRun {
   t.Fatalf("Expected CanRun to return %v when running %v on host %v", tc.shouldRun, tc.guestOs,  tc.host)
```

That would remove the switch cases here.